src/client/actions/user/index.ts

```diff
@@ -184,7 +186,7 @@ const createUrl = (dispatch: AllThunkDispatch, user: UserState) => {
     }
     return response.json().then((json) => {
       if (response.status === 200) {
-        dispatch<ResetUserStateAction>(resetUserState())
+        dispatch(resetUserState())
```

Comment:
what happens when the converse approach is done btw? like instead, we keep `dispatch<ResetUserStateActions>` here, but the argument of the function instead of `dispatch<AllActions>` we use a more specific etc

Reply:
Would it be too troublesome to have to change it if more actions are added? Since it is valid to dispatch any allowable action

Reply:
this is an interesting point. 

my interpretation of typescript integration in `dispatch` would be that we restrict ourselves to the smallest subset of types needed and work with that. This way we know that our function was designed to work with those specific types only, and if the `dispatch` calls inside the function body uses extra types it would throw, telling us this was not the intended use of that function.

paging @liangyuanruo for any inputs in this discussion

Reply:
wholly agree that we should narrow down the types and lean on the compiler for downstream type safety. dispatch doesn't know the type beforehand, so it's important to template the specific type in.